internal/tools/pipeline/dbclient/op_pipeline_labels.go

```diff
@@ -196,23 +196,29 @@ func (client *Client) SelectTargetIDsByLabels(req apistructs.TargetIDSelectByLab
 
 	if len(req.AnyMatchLabels) > 0 {
 		var allNeedMergeTargetIDs [][]uint64
+		innerSegments := append(sqlSegments, " AND (")
```

Comment:
maybe you should add unit tests for how to assemble `innerSegments`

Reply:
Return to the original condition